src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSource.cs

```diff
@@ -22,27 +21,25 @@ namespace NuGet.Protocol
 {
     public class HttpSource : IDisposable
     {
-        public static readonly TimeSpan DefaultDownloadTimeout = TimeSpan.FromSeconds(60);
-        public static readonly TimeSpan DefaultRequestTimeout = TimeSpan.FromSeconds(100);
         private const int BufferSize = 8192;
         private readonly Func<Task<HttpHandlerResource>> _messageHandlerFactory;
         private readonly Uri _baseUri;
         private HttpClient _httpClient;
         private string _httpCacheDirectory;
         private readonly PackageSource _packageSource;
+        private readonly IThrottle _throttle;
 
         // Only one thread may re-create the http client at a time.
         private readonly SemaphoreSlim _httpClientLock = new SemaphoreSlim(1, 1);
 
-        /// <summary>The timeout to apply to <see cref="DownloadTimeoutStream"/> instances.</summary>
-        /// <summary>This API is intended only for testing purposes and should not be used in product code.</summary>
-        public TimeSpan DownloadTimeout { get; set; } = DefaultDownloadTimeout;
-
         /// <summary>The retry handler to use for all HTTP requests.</summary>
         /// <summary>This API is intended only for testing purposes and should not be used in product code.</summary>
         public IHttpRetryHandler RetryHandler { get; set; } = new HttpRetryHandler();
 
-        public HttpSource(PackageSource packageSource, Func<Task<HttpHandlerResource>> messageHandlerFactory)
+        public HttpSource(
+            PackageSource packageSource,
+            Func<Task<HttpHandlerResource>> messageHandlerFactory,
+            IThrottle throttle = null)
```

Comment:
Not a big fan of default null arguments. Here is the deal. `HttpSourceResouceProvider` has all the necessary inputs to make the right decision about the throttle instance to use. And actually it's its sole purpose to build fully initialized resource instance. I'd suggest moving this logic over there while placing explicit null check here. Does it make sense?